packages/api-client/src/views/Request/ResponseSection/ResponseSection.vue

```diff
@@ -59,9 +59,54 @@ const activeSection = ref<ActiveSections>('All')
 
 /** Threshold for virtualizing response bodies in bytes */
 const VIRTUALIZATION_THRESHOLD = 200_000
-const shouldVirtualize = computed(
-  () => (props.response?.size ?? 0) > VIRTUALIZATION_THRESHOLD,
-)
+const shouldVirtualize = computed(() => {
+  if (!props.response) return false
+
+  // Get content type from headers
+  const contentType =
+    props.response.headers?.['content-type'] ||
+    props.response.headers?.['Content-Type']
+
+  // If no content type or response size is small, don't virtualize
+  if (!contentType || (props.response.size ?? 0) <= VIRTUALIZATION_THRESHOLD) {
+    return false
+  }
+
+  // Common text-based content types
+  const textBasedTypes = [
```

Comment:
This list already exists with the media type parser 😅

Reply:
so nice do it twice 😅

Reply:
I think we can just reference the raw ones here, also this mediaConfig should be set somewhere already I think

https://github.com/scalar/scalar/blob/main/packages/api-client/src/views/Request/consts/mediaTypes.ts